code/src/cljs/sixsq/nuvla/ui/deployment/views.cljs

```diff
@@ -2,946 +2,212 @@
+
+
+(defn deployments-display
+  [deployments-list]
+  (let [loading? (subscribe [::subs/loading?])
+        view     (subscribe [::subs/view])]
+    (fn [deployments-list]
+      [ui/Segment (merge style/basic
+                         {:loading @loading?})
+       (if (= @view "cards")
+         [cards-data-table deployments-list]
+         [vertical-data-table deployments-list])])))
+
+
+(defn StatisticStates
+  ([] [StatisticStates true])
+  ([clickable?]
+   (let [tr          (subscribe [::i18n-subs/tr])
+         summary     (subscribe [::subs/deployments-summary])
+         summary-all (subscribe [::subs/deployments-summary-all])
+         open-popup  (r/atom true)]
+     (fn [clickable?]
+       (let [summary       (if clickable? summary summary-all)
+             terms         (utils-general/aggregate-to-map (get-in @summary [:aggregations :terms:state :buckets]))
+             started       (:STARTED terms 0)
+             starting      (:STARTIN terms 0)
+             created       (:CREATED terms 0)
+             stopped       (:STOPPED terms 0)
+             error         (:ERROR terms 0)
+             queued        (:QUEUED terms 0)
+             starting-plus (+ starting created queued)
+             total         (:count @summary)
+             ]
+         [:div {:style {:margin     "10px auto 10px auto"
+                        :text-align "center"
+                        :width      "100%"}}
+          [ui/StatisticGroup (merge {:widths (if clickable? nil 5) :size "tiny"}
+                                    {:style {:margin-right "0px"
+                                             :display      "block"}})
+           [main-components/StatisticState total ["fas fa-rocket"] "TOTAL" clickable?
+            ::events/set-state-selector ::subs/state-selector]
+           [main-components/StatisticState started [(utils/status->icon utils/status-started)] utils/status-started
+            clickable? "green"
+            ::events/set-state-selector ::subs/state-selector]
+           [main-components/StatisticState starting-plus [(utils/status->icon utils/status-starting)]
+            utils/status-starting clickable? "yellow"
+            ::events/set-state-selector ::subs/state-selector]
+           [main-components/StatisticState stopped [(utils/status->icon utils/status-stopped)] utils/status-stopped
+            clickable? "yellow"
+            ::events/set-state-selector ::subs/state-selector]
+           (if clickable?
+             [ui/Popup {:trigger  (r/as-element
+                                    [main-components/StatisticState error [(utils/status->icon utils/status-error)]
+                                     utils/status-error clickable? "red"
+                                     ::events/set-state-selector ::subs/state-selector])
+                        :open     @open-popup
+                        :position "right center"
+                        :offset   [0 20]}
```

Comment:
Missing following:
`:style    {:z-index "auto"}                  ;to avoid pop up floating above modals`

The help can be extracted and reused to avoid this duplicated.
What about using this instead:
```
[:<>
              [ui/Statistic
               {:size     "tiny"
                :class    "slight-up"
                :style    {:cursor "pointer"}
                :on-click #(when clickable? (do (reset! show-state-statistics (not @show-state-statistics))
                                                (when-not @show-state-statistics
                                                  (dispatch [::events/set-state-selector nil]))))}
               [ui/StatisticValue {:style {:margin "0 10px"}}
                [ui/Icon {:name (if @show-state-statistics "angle double up" "angle double down")}]]]
              [ui/Label {:pointing "left" :basic true}
               [:span [ui/Icon {:name "arrow left"}] (@tr [:statistics-select-info])]]
              ]
```

Reply:
Yes. Extracted popup as a function (main-components/ClickMeStaticPopup) on a self contained span element.